### PR TITLE
Move all support for the legacy API into a plugin

### PR DESCRIPTION
**Pull request by akhenry**
closes https://github.com/nasa/openmct/issues/4615

This PR renders support for Open MCT's legacy API optional by moving the entire adapter layer into a plugin. 

Legacy support can be enabled by installing the new legacy support plugin like so:

``` JavaScript
openmct.install(openmct.plugins.LegacySupport());
```

If you also require support for legacy indicators you will also need to install the LegacyIndicatorsPlugin -

``` JavaScript
openmct.install(openmct.plugins.LegacyIndicators());
```

For now it remains installed by default in the Open MCT source. [A followup issue](https://github.com/nasa/openmct/issues/4613) has been filed to ensure that legacy support is disabled by default in 1.8.3.

In addition to moving legacy support into a plugin, this PR also:
* Adds a composition policy that was missing from the modern composition API to prevent composition of objects that do not support composability.
* Fixes a memory leak in timers that was causing problems in tests.
* Fixes a ton of tests broken by the removal of legacy code.

### All Submissions:

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [X] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [X] Changes address original issue?
* [X] Unit tests included and/or updated with changes?
* [X] Command line build passes?
* [X] Has this been smoke tested?
* [x] Testing instructions included in associated issue?

### CODE REVIEW

**Comment by unlikelyzero** [comment]
Note: This is passing our smoke tests.

**Review by davetsay — CHANGES_REQUESTED** [review]
This is great @akhenry . I ran this in VISTA and there were no issues.

There are a couple of minor change requests inline.

File: src/plugins/legacySupport/plugin.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+import installDefaultBundles from './installDefaultBundles';
+import BundleRegistry from './BundleRegistry';
+import Main from '../../../platform/framework/src/Main';
+import objectUtils from '../../api/objects/object-utils';
+import DomainObjectImpl from '../../../platform/core/src/objects/DomainObjectImpl';
+import ContextualDomainObject from '../../../platform/core/src/capabilities/ContextualDomainObject';
+
+export default function CouchPlugin(options) {
```
[the remark this answers is not in the document]
Comment (akhenry) [reply src/plugins/legacySupport/plugin.js:29]: Hehehe, it suuuuuure is. Thanks!

File: src/plugins/plugins.js [hunk 2]
```diff
@@ -237,6 +241,8 @@ define([
     plugins.Timer = Timer.default;
     plugins.DeviceClassifier = DeviceClassifier.default;
     plugins.LocalStorage = LocalStorage.default;
+    plugins.LegacySupport = LegacySupportPlugin.default;
+    plugins.LegacyIndicators = LegacyIndicatorsPlugin.default;
```
[the remark this answers is not in the document]
Comment (akhenry) [reply src/plugins/plugins.js:245]: Ok, lemme take a look. Thanks

[the remark this answers is not in the document]
Comment (akhenry) [reply src/plugins/plugins.js]: Good catch, fixed and smoke tested with `example/notifications`

File: src/plugins/legacySupport/plugin.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (akhenry) [reply src/plugins/legacySupport/plugin.js]: Fixed

**Review by davetsay — APPROVED** [review]
LGTM

## Reviewer Checklist
| | |
| --- | :---: |
| Changes appear to address issue? | Y |
| Appropriate unit tests included? | Y |
| Code style and in-line documentation are appropriate? | Y |
| Commit messages meet standards? | Y |
| Has associated issue been labelled `unverified`? | Y |

**Pull request merged**